camply/search/search_going_to_camp.py

```diff
@@ -0,0 +1,212 @@
+        self._campground_object = campgrounds
+        self.weekends_only = weekends_only
+        assert (
+            any(
+                [
+                    campsites not in [[], None],
+                    campgrounds not in [[], None],
+                    recreation_area is not None,
+                ]
+            )
+            is True
+        )
+        self.campsites = make_list(campsites)
+        self.campgrounds = self._get_searchable_campgrounds()
+        self.equipment = self._get_equipment_category(equipment)
+
+    def _get_equipment_category(self, equipment: List[Tuple[str, int]]):
+        our_equipment_name, our_equipment_length = equipment[0]
+        our_equipment_length = int(our_equipment_length)  # TODO: gross
+        equipment_categories = self.campsite_finder.list_equipment_categories(
+            self._recreation_area_id[0]
+        )
+        matching_category_id = None
+        for eq_name, category in equipment_categories.items():
+            category_id = category["id"]
+            if eq_name.find(our_equipment_name) > 0:
+                if our_equipment_length == 0:
+                    return category_id
+                if (
+                    our_equipment_length > 0
+                    and our_equipment_length <= category["max_size"]
+                ):
+                    return category_id
+
+        logger.error("No equipment category found that matches your equipment")
+        exit(1)
+
+    def get_all_campsites(self) -> List[AvailableCampsite]:
+        """
+        Search for all campsites matching search criteria
+
+        Returns
+        -------
+        List[AvailableCampsite]
+        """
+        for campground in self.campgrounds:
+            sites = self.campsite_finder.list_site_availability(
+                campground,
+                self.search_window[0].start_date,
+                self.search_window[0].end_date,
+                self.equipment,
+            )
+            available_sites = []
+            for site in sites:
+                site_details = self.campsite_finder.get_site_details(
+                    self._recreation_area_id[0], campground.facility_id
+                )
+
+                nights = (
+                    self.search_window[0].end_date - self.search_window[0].start_date
+                ).days
+                start_dt = datetime.combine(self.search_window[0].start_date, time.min)
+                end_dt = datetime.combine(self.search_window[0].end_date, time.min)
+                rec_area = self.campsite_finder.rec_area_lookup(
+                    rec_area_id=self._recreation_area_id[0]
+                )
+                booking_url = self.campsite_finder.get_reservation_link(
+                    party_size=1,
+                    start_date=self.search_window[0].start_date,
+                    end_date=self.search_window[0].end_date,
+                    camp_area=campground,
```

Comment:
The campground's map_id should point at the most deeply nested map that contains the site to be booked. Currently the map contains the facility's top-most ID, which requires the user to click through the the appropriate section on the booking page. 

Fix this.